### PR TITLE
Multiple rotations

### PR DESCRIPTION
**Pull request by kroppt**
This adds support for rotating in two directions at once.

### CODE REVIEW

**Review by kropptrevor — APPROVED** [review]

File: modules/player/api_test.go [hunk 1]
```diff
@@ -136,44 +136,76 @@ func TestModuleHandleMovementEvent(t *testing.T) {
...
-				V: [3]float32{math.Sin(1.0 / 2), 0, 0},
+				V: [3]float64{math.Sin(1.0 / 2), 0, 0},
+			},
+		},
+		{
+			right: math.Pi / 2.0,
+			down:  math.Pi / 2.0,
+			rotation: mgl.Quat{
+				W: 1.0 / 2.0,
+				V: [3]float64{1.0 / 2.0, -1.0 / 2.0, -1.0 / 2.0},
```
Comment (kropptrevor) [inline modules/player/api_test.go:208]: I don't think this test case is correct. That rotation gives a result of `{1, 0, 0}`. See [wolframalpha](https://www.wolframalpha.com/input/?i=quaternion+%281%2F2%2C1%2F2%2C-1%2F2%2C-1%2F2%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%281%2F2%2C+-1%2F2%2C+1%2F2%2C+1%2F2%29).

**Review by kropptrevor — CHANGES_REQUESTED** [review]

File: modules/player/api_test.go (hunk 1 above)
Comment (kropptrevor) [reply modules/player/api_test.go:208]: Doing just the X rotation also gives `{1,0,0}`. See [wolframalpha](https://www.wolframalpha.com/input/?i=quaternion+%28sqrt%5B2%5D%2F2%2C0%2C-sqrt%5B2%5D%2F2%2C0%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%28sqrt%5B2%5D%2F2%2C+0%2C+sqrt%5B2%5D%2F2%2C+0%29). Clearly, our math is wrong in the Y direction.

Comment (kropptrevor) [reply modules/player/api_test.go:208]: [This ](https://www.wolframalpha.com/input/?i=quaternion+%281%2F2%2C-1%2F2%2C-1%2F2%2C-1%2F2%29+*+quaternion+%280%2C+0%2C+0%2C+-1%29+*+quaternion+%281%2F2%2C+1%2F2%2C+1%2F2%2C+1%2F2%29) is the result we want, which I computed without rotating the Y direction axis of rotation.

Comment (kroppt) [reply modules/player/api_test.go:208]: OK I understand. The result you give rotates the up vector from +y to +x, which is what we want.

**Review by kropptrevor — APPROVED** [review]

**Review by gregjohnson2017 — APPROVED** [review]

**Pull request merged**